source/gitlab/gitlab.go

```diff
@@ -176,31 +203,37 @@ func (g *Gitlab) Close() error {
 
 func (g *Gitlab) First() (version uint, er error) {
 	if v, ok := g.migrations.First(); !ok {
-		return 0, &os.PathError{Op: "first", Path: g.path, Err: os.ErrNotExist}
+		return 0, &os.PathError{Op: "first", Path: g.config.Path, Err: os.ErrNotExist}
 	} else {
 		return v, nil
 	}
 }
 
 func (g *Gitlab) Prev(version uint) (prevVersion uint, err error) {
 	if v, ok := g.migrations.Prev(version); !ok {
-		return 0, &os.PathError{Op: fmt.Sprintf("prev for version %v", version), Path: g.path, Err: os.ErrNotExist}
+		return 0, &os.PathError{Op: fmt.Sprintf("prev for version %v", version), Path: g.config.Path, Err: os.ErrNotExist}
 	} else {
 		return v, nil
 	}
 }
 
 func (g *Gitlab) Next(version uint) (nextVersion uint, err error) {
 	if v, ok := g.migrations.Next(version); !ok {
-		return 0, &os.PathError{Op: fmt.Sprintf("next for version %v", version), Path: g.path, Err: os.ErrNotExist}
+		return 0, &os.PathError{Op: fmt.Sprintf("next for version %v", version), Path: g.config.Path, Err: os.ErrNotExist}
 	} else {
 		return v, nil
 	}
 }
 
 func (g *Gitlab) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
+	g.ensureFields()
+
 	if m, ok := g.migrations.Up(version); ok {
-		f, response, err := g.client.RepositoryFiles.GetFile(g.projectID, m.Raw, g.getOptions)
+		g.getOptions = &gitlab.GetFileOptions{
```

Comment:
Again, don't set instance params in methods that shouldn't be modifying the instance